src/pages/ultimate.tsx

```diff
@@ -0,0 +1,256 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { css } from 'emotion';
+
+export enum Winner {
+  Nought,
+  Cross,
+  Draw,
+}
+
+export interface Cell {
+  player: Nullable<Winner>;
+}
+
+export interface Board {
+  cells: Cell[];
+}
+
+function calculateWinner(squares: Cell[]): Nullable<Winner> {
+  const lines = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+  for (const [a, b, c] of lines) {
+    if (
+      squares[a].player !== null &&
+      squares[a].player !== Winner.Draw &&
+      squares[a].player === squares[b].player &&
+      squares[a].player === squares[c].player
+    ) {
+      return squares[a].player;
+    }
+  }
+  return squares.every(cell => cell.player !== null) ? Winner.Draw : null;
+}
+
+export interface GameState {
+  available: number[];
+  boards: Board[];
+  xIsNext: boolean;
+}
+
+export class Game extends React.Component<{}, GameState> {
+  constructor(props: {}) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      available: _.range(9),
+      boards: _.times(9, i => ({
+        cells: _.times(9, j => ({ player: null })),
+      })),
+      xIsNext: true,
+    };
+  }
+
+  handleClick(i: number, j: number) {
+    const boards = this.state.boards.slice();
+    boards[i] = {
+      cells: boards[i].cells.slice(),
+    };
+    if (
+      calculateWinner(boards[i].cells) !== null ||
+      boards[i].cells[j].player !== null
+    ) {
+      return;
+    }
+    boards[i].cells[j].player = this.state.xIsNext
+      ? Winner.Cross
+      : Winner.Nought;
+    let available: number[];
+    if (
+      calculateWinner(

```

Comment:
This can definitely be tidied a bit using some combination of variables (`overallWinner` for the outer if), `_.cond`, object literals, etc. Specifics up to you.